tests/sequence_rooter.rs

```diff
@@ -0,0 +1,74 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#![cfg(feature = "debugmozjs")]
+
+#[macro_use]
+extern crate mozjs;
+use mozjs::jsapi::JSTracer;
+use mozjs::rust::Runtime as Runtime_;
+use mozjs::rust::CustomTrace;
+use mozjs::jsapi::JS_GC;
+use mozjs::jsapi::RootKind;
+use mozjs::rust::RootKind as RootKind_;
+use mozjs::rust::GCMethods;
+
+/// Check if Rust reimplementation of CustomAutoRooter properly appends itself
+/// to autoGCRooters stack list and if C++ inheritance was properly simulated
+/// by checking if appropriate virtual trace function was called.
+#[test]
+fn custom_auto_rooter_vftable() {
+    static mut TRACE_FN_WAS_CALLED: bool = false;
+
+    pub struct TestStruct { }
+
+    unsafe impl CustomTrace for TestStruct {
+        fn trace(&self, _: *mut JSTracer) {
+            unsafe { TRACE_FN_WAS_CALLED = true; }
```

Comment:
We could store a `Cell<bool>` inside CustomTrace instead of using `static mut` for this.

Reply:
As we talked, `#[cfg(test)]` code is not compiled in a library target that's used by a Cargo integration test, so we can't modify the `CustomTrace` trait here.

I can extract the boilerplate code to a macro that declares the static variable and the impl block, if that works.

Reply:
Whoops, I meant to says `TestStruct` instead of `CustomTrace`.